actors/market/src/lib.rs

```diff
@@ -926,13 +942,14 @@ impl Actor {
                     // https://github.com/filecoin-project/builtin-actors/issues/1389
                     // handling of legacy deals is still done in cron. we handle such deals here and continue to
                     // reschedule them. eventually, all legacy deals will expire and the below code can be removed.
-                    let (slash_amount, _payment_amount, remove_deal) = st.process_deal_update(
-                        rt.store(),
-                        &state,
-                        &deal_proposal,
-                        &dcid,
-                        curr_epoch,
-                    )?;
+                    let (slash_amount, _payment_amount, completed, remove_deal) = st
+                        .process_deal_update(
+                            rt.store(),
+                            &state,
+                            &deal_proposal,
+                            &dcid,
+                            curr_epoch,
+                        )?;
 
                     if remove_deal {
```

Comment:
I think this should emit deal-terminated, immediately after the `provider_deals_to_remove` update at 963. This will mimic the same code in the synchronous case. Yes, this is inside cron, but the FIP says we will emit the event here.

Reply:
Added the corresponding here but note that it will only be emitted if the deal is not already completed as we don't want to emit both "completed" and "terminated" events in this case.